source/features/branch-buttons.tsx

```diff
@@ -93,7 +93,7 @@ async function getDefaultBranchLink() {
 	);
 }
 
-async function init() {
+async function init(): Promise<boolean | void> {
```

Comment:
The `Promise<boolean | void>` return type allows us in strict mode to not return in all code paths as the code currently does. I'll only mention this once as this repeats for many `init()`s.

Reply:
We have `featureFunction` and `featureFunctionPromised` (can they be merged?)

Reply:
Actually, from what I can tell, they can't as one has a Promise as a return type and the other is not. I don't think there is a way to infer if a function is async or not. I'll dig a little to see.

Reply:
The Feature details already has a union type

```
interface FeatureDetails {
    ...
	init: FeatureFunction | FeaturePromisedFunction;
	...
}
```

and those types just got tweaked a bit.

```
export type FeatureInit = boolean | void;
export type AsyncFeatureInit = Promise<FeatureInit>;
type FeatureFunction = () => AsyncFeatureInit;
type FeaturePromisedFunction = () => AsyncFeatureInit;
```

So now I can use `FeatureInit` or `AsyncFeatureInit` on the `init()` functions of features. And just in case it was lost in the slew of comments, we need an explicit return type for two reasons. One, because of the lint rule, but also because all the feature `init()` function have a pattern of returning `false` or `undefined` (when no explicit return, which TypeScript doesn't like).

![image](https://user-images.githubusercontent.com/833231/53316969-cb16d880-3897-11e9-93ac-d6ad3efe656e.png)

A union on the return type of the `init()` functions is not possible, because putting an explicit return type when the function is `async` will not allow for this basically `boolean | void | Promise<boolean | void>`

![image](https://user-images.githubusercontent.com/833231/53316863-6fe4e600-3897-11e9-8893-4ec4a8740cdd.png)

Reply:
I had to discard these types due to https://github.com/benmosher/eslint-plugin-import/issues/1282

I think the result isn't bad at all:

```ts
import features from '../libs/features';
async function init(): Promise<boolean | void> {
```

instead of:

```ts
import features, {AsyncFeatureInit} from '../libs/features';
async function init(): AsyncFeatureInit {
```